pages/index.js

```diff
@@ -1,15 +1,28 @@
 import Head from "next/head";
-import Header from "@/components/Header";
-import ActivityList from "@/components/ActivityList";
+import Header from "@/components/layout/Header";
+import useLocalStorageState from "use-local-storage-state";
```

Comment:
It's better to have the state in the _app.js and pass it as a prop to all your pages. Your solution could work fine at the moment although it has some issues: For example if you try to access `activities/2` with an empty local storage you won't find any activity. And even more: Your local storage will remain empty even if you visit index.js afterwards.